File: week5_swappin_aint_easy/solutions/swap/Swap.java
public class Swap {

    // Java is always pass by value
    public static int[] swapByValue(int a, int b) {
        return new int[] {b, a};  // Return an array with swapped values
    }

    public static void main(String[] args) {
        int a = 5, b = 3;

        System.out.println("Before: a=" + a + ", b=" + b);
        int[] swapped = swapByValue(a, b);
        System.out.println("After swapByValue: a=" + swapped[0] + ", b=" + swapped[1]);
    }
}

File: challenges/week5_swappin_aint_easy/solutions/swap/swap.php
<?php
function swapViaArr($a, $b) {
    return array($b, $a);
}

function swapByXorReference(&$a, &$b) {
    $a = $a ^ $b;
    $b = $a ^ $b;
    $a = $a ^ $b;
}

$a = 5; $b = 3; 
echo "Before: a=$a, b=$b\n";

list($a, $b) = swapViaArr($a, $b);
echo "After via Arr Swap: a=$a, b=$b\n";


swapByXorReference($a, $b);
echo "After xor by ref: a=$a, b=$b\n";
?>


